go/worker/registration/worker.go

```diff
@@ -941,30 +968,25 @@ func (w *Worker) registerNode(epoch beacon.EpochTime, hook RegisterNodeHook) err
 		nodeSigners = append([]signature.Signer{w.identity.NodeSigner}, nodeSigners...)
 	}
 
-	sigNode, err := node.MultiSignNode(nodeSigners, registry.RegisterNodeSignatureContext, &nodeDesc)
-	if err != nil {
+	sigNode, grr := node.MultiSignNode(nodeSigners, registry.RegisterNodeSignatureContext, &nodeDesc)
+	if grr != nil {
 		w.logger.Error("failed to register node: unable to sign node descriptor",
-			"err", err,
+			"err", grr,
 		)
-		return err
+		err = fmt.Errorf("unable to sign node descriptor: %w", grr)
+		return
 	}
 
 	tx := registry.NewRegisterNodeTx(0, nil, sigNode)
-	if err := consensus.SignAndSubmitTx(w.ctx, w.consensus, w.registrationSigner, tx); err != nil {
+	if err = consensus.SignAndSubmitTx(w.ctx, w.consensus, w.registrationSigner, tx); err != nil {
 		w.logger.Error("failed to register node",
 			"err", err,
 		)
-		return err
+		return
 	}
 
-	// Update the registration status on successful registration.
-	w.RLock()
-	w.status.LastRegistration = time.Now()
-	w.status.Descriptor = &nodeDesc
-	w.RUnlock()
-
 	w.logger.Info("node registered with the registry")
-	return nil
+	return
```

Comment:
```suggestion
	return nil
```
For sake of clarity I prefer explicit return statements (e.g. `return err`) even if return values are named.